src/cli.rs

```diff
@@ -949,6 +957,41 @@ impl Pretty {
     }
 }
 
+#[derive(Debug)]
+pub struct FormatOptions {
+    pub json_indent: usize,
+}
+
+impl Default for FormatOptions {
+    fn default() -> Self {
+        Self { json_indent: 4 }
+    }
+}
+
+impl FromStr for FormatOptions {
+    type Err = anyhow::Error;
+    fn from_str(options: &str) -> anyhow::Result<FormatOptions> {
+        let mut format_options = FormatOptions::default();
+
+        // generate a map of the specified options
+        for argument in options.split(',') {
+            let argument_parts: Vec<&str> = argument.splitn(2, ':').collect();
+            match argument_parts[..] {
+                ["json.indent", value] => {
+                    format_options.json_indent =
+                        value.parse().map_err(|_| anyhow!("Invalid indent value"))?;
+                }
+                _ => {
+                    return Err(anyhow!(
+                        "Format options consist of a key and a value, separated by a \":\"."
```

Comment:
I think we should have different error messages for a missing `:` and an unknown key.

And since `json.sort_keys` exists in HTTPie maybe it can get a special message explaining that it's unimplemented?

Reply:
Ah, it looks like HTTPie has more options besides those:
- `headers.sort`
- `json.format`
- `json.sort_keys`
- `xml.format`
- `xml.indent`

It would be nice to hardcode that list for a different error message but if you want to skip that then that's fine by me.

Reply:
I plan to look at the remaining options in a follow-up PR so we can leave the error as it is now.